### PR TITLE
Firelocks are now sentient + better firealarms

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

This essentially disconnects fire doors from the fire alarm network and enables each individual fire door to check the adjacent atmosphere using a signal. So in theory this won't lag.

The result of this is that fire alarms no longer GBJ everyone when there is no atmosphere danger, imagine you're in a hallway, and on the other side of the hallway, a fire is raging. Yet your perfectly safe hallway decides to GBJ you and close ALL the firelocks. 

Firelocks now display a HOT/HIGH PRESSURE or COLD/LOW PRESSURE warning via a small flashing light on the side of the door. 

An automatic trigger of the alarm will only trigger the sounds and lights of the area.

A manual trigger of the alarm will trigger all of the above AND close firedoors. Turning the alarm OFF will open all firedoors.

https://user-images.githubusercontent.com/9026500/123877654-830b9300-d935-11eb-9376-0989562678c4.mp4

Active opening test:

https://user-images.githubusercontent.com/9026500/123877875-ee556500-d935-11eb-9e8f-3343919c21e3.mp4

## Changelog
:cl:
add: Sentient firelocks.
/:cl:

### CODE REVIEW

File: code/game/machinery/doors/firedoor.dm [hunk 1]
```diff
@@ -44,19 +44,25 @@
 	else
 		. += span_notice("The bolt locks have been <i>unscrewed</i>, but the bolts themselves are still <b>wrenched</b> to the floor.")
 
+/obj/machinery/door/firedoor/proc/CalculateAffectingAreas() //SKYRAT EDIT CHANGE
```
Comment (FlamingLily) [inline code/game/machinery/doors/firedoor.dm]: Couldn't this be put in modular? We already have a firedoor.dm in master files

File: code/game/machinery/doors/firedoor.dm [hunk 2]
```diff
@@ -206,7 +213,10 @@
 
 /obj/machinery/door/firedoor/update_icon_state()
 	. = ..()
-	icon_state = "[base_icon_state]_[density ? "closed" : "open"]"
```
Comment (FlamingLily) [inline code/game/machinery/doors/firedoor.dm:209]: Non-modular comments

File: code/game/machinery/doors/firedoor.dm (hunk 1 above)
Comment (Gandalf2k15) [reply code/game/machinery/doors/firedoor.dm]: Wait for the updates

**Comment by Weblure** [comment]
Thank you for making this, our current system is hell

**Comment by Deek-Za** [comment]
After getting some time with this gotta say it's a great change so far. Those old fire alarms were just awful and created more issues then they helped prevent.

**Pull request merged**